### ISSUE TITLE
This tool is highly inaccurate and does not work

### ISSUE THREAD

**Issue by Bubbachunk** [cadon/ARKStatsExtractor#536]
I have made every adjustment and changed the settings accordingly and it always fails, never reads the screen or screenshots and always fails when info is entered manually. DO NOT INSTALL, WASTE OF TIME. Wasted 2 months trying to get it to work.

**Comment by Wardjp** [comment]
Dear lord, way to leave a pointless review in the bug section?... This quite shows how likely this entire complaint is your own fault. I and everyone I know that cares about their taming and breeding programs have been using this program for years. I can assure you it is more accurate than anything else out there and that if you have a problem, you should be asking for help instead of aimlessly slandering a developer that has been working his ass off for us for little to no compensation, to create what is the most powerful application in all of Ark.

**Comment by Bubbachunk** [comment]
Actually, I have not had to ask for help because I am having the same issues as a lot of the people here, which always ends up with the same conclusion but no fix.......The program itself does not add up or compensate for imprinting correctly or ARK itself does not add up or compensate for imprinting correctly. I am a little confused about the part where you say a developer has worked his ass off for us, begging your pardon, but no developer that actually "works" on a program like this, offers it for free, he has very much enjoyed creating this program and puts it out there as a courtesy, I understand that, but don't be fooled thinking he "works" at it, it is fun for him, not work. Very similar to you tubers and Twitch streamers who "very much enjoy" what they do and it is not work for them. I appreciate the developer's attempt at this program, but it has wasted 2 months of playtime for me, just trying to get it to work and spending endless hours contacting my server owner and getting numbers for the three different servers I play on all to no avail. Ironically, it has turned "my fun playtime", into what feels very much like work. I have a real job, and over the last 3 years have poured over 2,170 hours into ARK, it is my funtime, not work away from work.

**Comment by cadon** [comment]
To be fair, your first comment sounds pretty pissed off, if you excuse my language. I understand this, trying to get something to work for that long is annoying. The stat-system of ARK is complex, it depends on a lot of multipliers that affect each other, and the slightest error in one of them can screw everything up. The tool is very restrictive, and refuses to let you save values if there a small error. The tool is designed like this because with the stats it's rather an "all good" or "probably a lot of the extracted data is pretty much off".

Theoretically the extraction is straight forward, but the game has some issues / bugs that makes the extraction hard. A common issue, the torpor-bug (displaying a torpor way too high right after taming) was fixed some months ago, which made it a lot easier for the extraction. A persisting issue is the imprinting. The game seems to forget to increase some stats after a cuddle and seems to apply this increase retroactively later (after a server-restart, going out of render-distance? I don't know exactly). Thus it's recommend to extract a creature before starting the imprinting.
Another issue that makes the extraction near impossible are creatures that are partially raised during breeding-events, causing different multipliers to be valid during their maturation.

I'm always open to new ideas how to solve these issues. I once thought it's impossible to extract imprinted creatures, because the game only shows the rounded numbers but internally calculates with decimal numbers until someone pointed out that the imprinting increase depends on the maturation-time and so the decimal numbers can be guessed by the tool. Maybe there's an easy way to compensate for different maturation-times with events as well and I just haven't found a good solution for that. Another thing is, theoretically a solution may work, but the game itself has to follow that as well. I guess performance is often a reason to cache values rather than recalculate them often enough to always show the true values.

Back to your criticism, you're right that it's fun for me to develop this tool, even more when I see it's used by and helps a lot of people. I'm aware that there are issues that prevent the tool from working, some might be bugs, e.g. caused by server-settings screwing up the calculations. A big issue, especially for multipliers or settings that differ a lot from the default settings, are rounding-errors. As said, the game only shows rounded numbers, but in reality calculates with a lot of decimals (not only imprinting, stats as well). It's a balance between allowing large deviations as still correct and keeping the error-range small enough to not get wrong results. I improved that system some time ago by making this range dynamically in some cases, but there might be cases still not be covered with that.

Btw, did you ever submit an issue with a creature that doesn't work? It's often cases like that that leads to resolving bugs in the tool.

And @Wardjp, thanks for the kind words!

**Comment by Wardjp** [comment]
Anytime, I've been using this app for a very very long time, I still remember stumbling upon it way back in the day when my tribe played on the Ark of War unofficial server, one of the top servers. It was so frustrating back then because we could never get accurate numbers to fill in the text file for the multipliers, and it was even more confusing just trying to accurately translate what we were given into that file. You have brought this app a LONG way from day 1 and that's what makes it incredible. I can tell you from experience amongst many longstanding alpha tribes, this is their go-to tool for long-term top of the line breeding. Heck, the tool actually has multiple avenues that people don't even realize. If an unofficial admin adds one of the binocular add-ons that show W/D levelups, you can actually use Stat Testing to input their values, no matter how screwed up the backend numbers might've been.

**Comment by Bubbachunk** [comment]
I understand where you are coming from and yes, I was very pissed at the time I made the post, I also understand the problems with the variation decimal numbers can cause. However, I play on an unofficial server in which a nanny is in use, the imprinting is always exactly 100%, which should actually not cause an issue at all, furthrmore, the base numbers will change after taming and imprinting a great deal, but, this should not even be a factor when the taming is always 100% TE and the imprinting is always 100% as well. so maybe now you understand why I felt the tool was broken and I work everyday, when I come home with the intention of getting in some game time, it is very frustrating to spend virtually all that time trying to get it to work when I am already fairly good at eyeballing it, if you know what I mean. So please, continue to develop the tool, but try to figure out why, when the settings are correct, that a full 100% TE creature that is also 100% imprinted, fails to extract, that really baffles me. Sorry for posting such a harsh post, but i was really frustrated and working 12 hr day shifts on 4 hours sleep for 3 weeks did not help.

**Comment by cadon** [comment]
When a creature hits 100% imprinting, the actual value (which can be a bit higher than 100%) is cut off and set to exactly 100% ingame. Usually, 100% isn't a value that is valid, e.g. a creature that gains 7% for each cuddle could have 98% or 105% imprinting, but not 100%. To consider the cutoff, there's a checkbox below the imprinting ("Exact, don't adjust") which usually is only needed if a creature has exactly 100%, so the tool doesn't guesses the next possible value (in the example 98%). You might ask, why you have to set the checkbox, the tool could just stop guessing if 100% is entered, and that was the case in the beginning. Until users said, that's an issue with certain creatures, e.g. the giganotosaurus, which could display 100% imprinting, but internally only have 99.7% imprinting. So this pesky checkbox had to be introduced (which causes some issues with users using it wrong, but for some species the tool just can't know what is the case).

The best would be, if you could post the stats of a creature that isn't working, I'm sure there are things that could be improved regarding this (on official servers rare) case of 100% taming / imprinting.

@Wardjp the superspyglass mod is super useful, especially in the cases the game does show wrong values, it helped to figure out some of the oddities the game does with the cached and wrong stat-values.

**Comment by Bubbachunk** [comment]
I have and use the super spyglass, I will post a screenshot soon.

**Comment by Bubbachunk** [comment]
ooops, didnt mean to close it.
